tools/flaky-test-retryer/github_commenter_test.go

```diff
@@ -0,0 +1,191 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/google/go-github/github"
+	"github.com/knative/test-infra/shared/ghutil/fakeghutil"
+	"github.com/knative/test-infra/tools/monitoring/prowapi"
+)
+
+var (
+	oldCommentBody = `<!--AUTOMATED-FLAKY-RETRYER-->
+The following tests are currently flaky. Running them again to verify...
+
+Test name | Retries
+--- | ---
+fakejob0 | 0/3
+fakejob1 | 1/3
+
+Automatically retrying...
+/test fakejob1`
+	retryCommentBody = `<!--AUTOMATED-FLAKY-RETRYER-->
+The following tests are currently flaky. Running them again to verify...
+
+Test name | Retries
+--- | ---
+fakejob0 | 1/3
+fakejob1 | 1/3
+
+Automatically retrying...
+/test fakejob0`
+	noMoreRetriesCommentBody = `<!--AUTOMATED-FLAKY-RETRYER-->
+The following tests are currently flaky. Running them again to verify...
+
+Test name | Retries
+--- | ---
+fakejob0 | 3/3
+fakejob1 | 1/3
+
+`
+	failedShortCommentBody = `<!--AUTOMATED-FLAKY-RETRYER-->
+The following tests are currently flaky. Running them again to verify...
+
+Test name | Retries
+--- | ---
+fakejob0 | 0/3
+fakejob1 | 1/3
+
+Failed non-flaky tests preventing automatic retry of fakejob0:
+
+`+"```\ntest0\ntest1\ntest2\ntest3\n```"
+	failedLongCommentBody = `<!--AUTOMATED-FLAKY-RETRYER-->
+The following tests are currently flaky. Running them again to verify...
+
+Test name | Retries
+--- | ---
+fakejob0 | 0/3
+fakejob1 | 1/3
+
+Failed non-flaky tests preventing automatic retry of fakejob0:
+
+`+"```\ntest0\ntest1\ntest2\ntest3\ntest4\ntest5\ntest6\ntest7\n```\n\nand 2 more."
+
+	fakeOrg       = "fakeorg"
+	fakeRepo      = "fakerepo"
+	fakeUserLogin = "fakelogin"
+	fakePullID    = 127
+	fakeCommentID = int64(1)
+	fakeUserID    = int64(99)
+	fakeUser      = &github.User{
+		ID:    &fakeUserID,
+		Login: &fakeUserLogin,
+	}
+	fakeOldComment = &github.IssueComment{
+		ID:   &fakeCommentID,
+		Body: &oldCommentBody,
+		User: fakeUser,
+	}
+	fakeJob = JobData{
+		&prowapi.ReportMessage{
+			JobName: "fakejob0",
+			Refs: []prowapi.Refs{{
+				Org:  fakeOrg,
+				Repo: fakeRepo,
+				Pulls: []prowapi.Pull{{
+					Number: fakePullID,
+				}},
+			}},
+		},
+		nil,
+		nil,
+	}
+
+	fakeFailedTests = []string{"test0", "test1", "test2", "test3", "test4", "test5", "test6", "test7", "test8", "test9"}
+)
+
+func getFakeGithubClient() *GithubClient {
+	gc := fakeghutil.NewFakeGithubClient()
+	gc.Repos = []string{fakeRepo}
+	gc.User = fakeUser
+	return &GithubClient{
+		gc,
+		*gc.User.Login,
+		false,
+	}
+}
+
+func TestGetOldComment(t *testing.T) {
+	gc := getFakeGithubClient()
+	cases := []struct {
+		client      *GithubClient
+		org, repo   string
+		pull        int
+		wantComment *github.IssueComment
+		wantErr     error
+	}{
+		{gc, fakeOrg, fakeRepo, fakePullID, nil, nil},
```

Comment:
nit, this test is different from what most table-driven tests, since the order matters, you may want to add a comment here

Reply:
gotcha, added a comment